### ISSUE TITLE
Hitting "Whoops, looks like something went wrong." after module installation; recurs automatically after fix.

### ISSUE THREAD

**Issue by SFFLAB** [freescout-helpdesk/freescout#820]
This morning I attempted to install the Customer Data Enrichment, Mobile Notifications, Two-Factor, and Reports module. As I was doing so, I hit the "Whoops, looks like something went wrong" error.

I had believe I identified the cause, which was that PHP curl was not installed properly. I deleted the modules via SSH, installed PHP curl (which now shows as OK by FreeScout), and reset the app cache. This brought the app back online and it behaves normally.

However, roughly an hour after doing this, the app fails again and displays the "Whoops, looks like something went wrong" error. I can reset the app cache again to fix it but obviously something is still wrong here...

Looking at the laravel log for today, I see the following error:

```
file_put_contents(/var/www/html/storage/framework/cache/data/5e/3c/5e3cbbed47195384f39edf30b47d8cd8245cdff8): failed to open stream: Permission denied {"exception":"[object] (ErrorException(code: 0): file_put_contents(/var/www/html/storage/framework/cache/data/5e/3c/5e3cbbed47195384f39edf30b47d8cd8245cdff8): failed to open stream: Permission denied at /var/www/html/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php:122)
[stacktrace]
#0 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(2, 'file_put_conten...', '/var/www/html/v...', 122, Array)
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php(122): file_put_contents('/var/www/html/s...', '1603127496a:12:...', 2)
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/FileStore.php(65): Illuminate\\Filesystem\\Filesystem->put('/var/www/html/s...', '1603127496a:12:...', true)
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/Repository.php(195): Illuminate\\Cache\\FileStore->put('laravel-modules', Array, 60)
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/Repository.php(327): Illuminate\\Cache\\Repository->put('laravel-modules', Array, 60)
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(304): Illuminate\\Cache\\Repository->remember('laravel-modules', 60, Object(Closure))
#6 /var/www/html/overrides/Nwidart/Modules/Repository.php(237): Illuminate\\Cache\\CacheManager->__call('remember', Array)
#7 /var/www/html/overrides/Nwidart/Modules/Repository.php(179): Nwidart\\Modules\\Repository->getCached()
#8 /var/www/html/overrides/Nwidart/Modules/Repository.php(261): Nwidart\\Modules\\Repository->all()
#9 /var/www/html/overrides/Nwidart/Modules/Repository.php(289): Nwidart\\Modules\\Repository->getByStatus(1)
#10 /var/www/html/overrides/Nwidart/Modules/Repository.php(331): Nwidart\\Modules\\Repository->enabled()
#11 /var/www/html/overrides/Nwidart/Modules/Repository.php(363): Nwidart\\Modules\\Repository->getOrdered()
#12 /var/www/html/vendor/nwidart/laravel-modules/src/Providers/BootstrapServiceProvider.php(22): Nwidart\\Modules\\Repository->register()
#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(586): Nwidart\\Modules\\Providers\\BootstrapServiceProvider->register()
#14 /var/www/html/vendor/nwidart/laravel-modules/src/ModulesServiceProvider.php(38): Illuminate\\Foundation\\Application->register(Object(Nwidart\\Modules\\Providers\\BootstrapServiceProvider))
#15 /var/www/html/vendor/nwidart/laravel-modules/src/LaravelModulesServiceProvider.php(15): Nwidart\\Modules\\ModulesServiceProvider->registerModules()
#16 [internal function]: Nwidart\\Modules\\LaravelModulesServiceProvider->boot()
#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#19 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#20 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#21 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(792): Illuminate\\Container\\Container->call(Array)
#22 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(775): Illuminate\\Foundation\\Application->bootProvider(Object(Nwidart\\Modules\\LaravelModulesServiceProvider))
#23 [internal function]: Illuminate\\Foundation\\Application->Illuminate\\Foundation\\{closure}(Object(Nwidart\\Modules\\LaravelModulesServiceProvider), 22)
#24 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(776): array_walk(Array, Object(Closure))
#25 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/BootProviders.php(17): Illuminate\\Foundation\\Application->boot()
#26 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(213): Illuminate\\Foundation\\Bootstrap\\BootProviders->bootstrap(Object(Illuminate\\Foundation\\Application))
#27 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(162): Illuminate\\Foundation\\Application->bootstrapWith(Array)
#28 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(146): Illuminate\\Foundation\\Http\\Kernel->bootstrap()
#29 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(116): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#30 /var/www/html/public/index.php(64): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#31 {main}
"}
```

Is there a permissions issue causing the systematic crashing? I'm a bit lost mostly because I would have thought deleting the modules that caused this would have rectified the issue; that it recurs in this way is unusual. Thanks.

**Comment by diguinho2018** [comment]
Hey!
I had the exact same issue when I installed the Two Factor module as seen on #807, solved by doing a FreeScout fresh install and I didn't install the Two Factor module addon again.

**Comment by SFFLAB** [comment]
Thanks much for the insight @diguinho2018! That may be the source of the issue, as that seems to be a relatively new module.

In my case a fresh install really isn't practical, as multiple independent businesses rely on my implementation and it would be way too disruptive. So hoping to find a way to resolve this, and I'll likely just not install any of the modules I had intended to for the foreseeable future.

**Comment by freescout-helpdesk** [comment]
https://github.com/freescout-helpdesk/freescout/wiki/Installation-Guide#if-you-are-receiving-failed-to-open-stream-error-message

**Comment by SFFLAB** [comment]
Thanks @freescout-helpdesk - just to clarify; are you suggesting I re-run the perms commands and see if that fixes the issue?

**Comment by SFFLAB** [comment]
I ran:
```
sudo chgrp -R www-data storage bootstrap/cache public/css/builds public/js/builds
sudo chmod -R ug+rwx storage bootstrap/cache public/css/builds public/js/builds
```
Upon doing so, I also rebooted the instance. So far everything is stable and the laravel log is clean, with no recurrence of the "Something went wrong" issue. I'm cautiously optimistic that this has been fixed. 

I'll update if that isn't sustained and will likewise try to remember to report back by end of week if it's indeed fully resolved.

**Comment by diguinho2018** [comment]
> 
> 
> I ran:
> 
> ```
> sudo chgrp -R www-data storage bootstrap/cache public/css/builds public/js/builds
> sudo chmod -R ug+rwx storage bootstrap/cache public/css/builds public/js/builds
> ```
> 
> Upon doing so, I also rebooted the instance. So far everything is stable and the laravel log is clean, with no recurrence of the "Something went wrong" issue. I'm cautiously optimistic that this has been fixed.
> 
> I'll update if that isn't sustained and will likewise try to remember to report back by end of week if it's indeed fully resolved.

By now it solved the issue with Two Factor module?
Did you run this by SSH?
I might try to reinstall this module and just to take a note if anything goes wrong.

**Comment by SFFLAB** [comment]
@diguinho2018 I removed the module entirely, so this hasn't fixed it while maintaining the module; I've just gotten back to where our (stable) setup was before I had tried to install it.

As much as I would like to install the modules I mentioned in the original issue, stability with what we've got is far more important, so I will not be attempting this again for a while.

I did all of this via SSH, on an EC2 instance.
